core/config/config.go

```diff
@@ -20,201 +19,358 @@ import (
+		Name:     "base-config-file",
+		Required: true,
+		Usage:    "Load base configurations from `FILE`",
 	}
+	AggregatorConfigFileFlag = cli.StringFlag{
+		Name:     "aggregator-config-file",
+		Required: true,
+		Usage:    "Load aggregator configurations from `FILE`",
+	}
+	OperatorConfigFileFlag = cli.StringFlag{
+		Name:     "operator-config-file",
+		Required: true,
+		Usage:    "Load operator configurations from `FILE`",
+	}
+	// Optional Flags
+)
+
+func NewAggregatorConfig(baseConfigFilePath, aggregatorConfigFilePath string) (*AggregatorConfig, error) {
+
+	if _, err := os.Stat(baseConfigFilePath); errors.Is(err, os.ErrNotExist) {
+		log.Fatal("Setup base config file does not exist")
+	}
+
+	if _, err := os.Stat(aggregatorConfigFilePath); errors.Is(err, os.ErrNotExist) {
+		log.Fatal("Setup aggregator config file does not exist")
+	}
+
+	baseConfig, err := newBaseConfig(baseConfigFilePath)
 
-	logger, err := sdklogging.NewZapLogger(configRaw.Environment)
 	if err != nil {
-		fmt.Println("Could not initialize logger")
+		log.Fatal("Error reading base config: ", err)
 	}
 
-	var alignedLayerDeploymentRaw AlignedLayerDeploymentRaw
-	alignedLayerDeploymentFilePath := ctx.GlobalString(AlignedLayerDeploymentFileFlag.Name)
-	if _, err := os.Stat(alignedLayerDeploymentFilePath); errors.Is(err, os.ErrNotExist) {
-		logger.Errorf("Path does not exist", "path", alignedLayerDeploymentFilePath)
-		return nil, err
+	var aggregatorConfigFromYaml AggregatorConfig
+	err = sdkutils.ReadYamlConfig(aggregatorConfigFilePath, &aggregatorConfigFromYaml)
+
+	if err != nil {
+		log.Fatal("Error reading aggregator config: ", err)
 	}
-	err = sdkutils.ReadJsonConfig(alignedLayerDeploymentFilePath, &alignedLayerDeploymentRaw)
+
+	return &AggregatorConfig{
+		BaseConfig: baseConfig,
+		Aggregator: aggregatorConfigFromYaml.Aggregator,
+	}, nil
+
+}
+
+func NewOperatorConfig(baseConfigFilePath, operatorConfigFilePath string) (*OperatorConfig, error) {
+
+	if _, err := os.Stat(baseConfigFilePath); errors.Is(err, os.ErrNotExist) {
+		log.Fatal("Setup base config file does not exist")
+	}
+
+	if _, err := os.Stat(operatorConfigFilePath); errors.Is(err, os.ErrNotExist) {
+		log.Fatal("Setup operator config file does not exist")
+	}
+
+	baseConfig, err := newBaseConfig(baseConfigFilePath)
+
 	if err != nil {
-		logger.Errorf("Cannot read aligned layer deployment file", "err", err)
-		return nil, err
+		log.Fatal("Error reading base config: ", err)
 	}
 
-	ethRpcClient, err := eth.NewClient(configRaw.EthRpcUrl)
+	var operatorConfigFromYaml OperatorConfig
+	err = sdkutils.ReadYamlConfig(operatorConfigFilePath, &operatorConfigFromYaml)
+
 	if err != nil {
-		logger.Errorf("Cannot create http ethclient", "err", err)
-		return nil, err
+		log.Fatal("Error reading operator config: ", err)
+	}
+
+	return &OperatorConfig{
+		BaseConfig: baseConfig,
+		Operator:   operatorConfigFromYaml.Operator,
+	}, nil
+
+}
+
+func newBaseConfig(baseConfigFilePath string) (*BaseConfig, error) {
```

Comment:
If we are using log.Fatal instead of returning error, remove error from return type. Two options here:

Go back to return error, and log error instead of fatal (which kills process)
Return just *BaseConfig